### PR TITLE
[IMP] hr_contract: add ordering option to structure types

### PR DESCRIPTION
**Pull request by dasz-odoo**
added sequence + handle for tree view and updated compute to select company's country's structure type first.

task 3279050

### CODE REVIEW

File: addons/hr_contract/models/hr_contract.py [hunk 1]
```diff
@@ -95,23 +95,18 @@ def _compute_employee_contract(self):
...
-                '|',
-                ('country_id', '=', self.company_id.country_id.id),
-                ('country_id', '=', False)])
-            if structure_types:
-                self.structure_type_id = structure_types[0]
-            elif self.structure_type_id not in structure_types:
-                self.structure_type_id = False
+    @api.depends('company_id')
+    def _compute_structure_calendar(self):
+        StructureType = self.env['hr.payroll.structure.type']
```
Comment (sofiagvaladze) [inline addons/hr_contract/models/hr_contract.py:100]:
```suggestion
        structure_type = self.env['hr.payroll.structure.type']
```

File: addons/hr_contract/models/hr_contract.py [hunk 2]
```diff
@@ -95,23 +95,18 @@ def _compute_employee_contract(self):
...
-            structure_types = self.env['hr.payroll.structure.type'].search([
-                '|',
-                ('country_id', '=', self.company_id.country_id.id),
-                ('country_id', '=', False)])
-            if structure_types:
-                self.structure_type_id = structure_types[0]
-            elif self.structure_type_id not in structure_types:
-                self.structure_type_id = False
+    @api.depends('company_id')
+    def _compute_structure_calendar(self):
```
Comment (sofiagvaladze) [inline addons/hr_contract/models/hr_contract.py:99]:
```suggestion
    def _compute_structure_type_id(self):
```

Comment (dasz-odoo) [reply addons/hr_contract/models/hr_contract.py:99]: done

File: addons/hr_contract/models/hr_contract.py (hunk 1 above)
Comment (dasz-odoo) [reply addons/hr_contract/models/hr_contract.py:100]: this is to signify that I'm using this as the class, not the instance.

File: addons/hr_contract/models/hr_contract.py [hunk 3]
```diff
@@ -95,23 +95,18 @@ def _compute_employee_contract(self):
...
-            if structure_types:
-                self.structure_type_id = structure_types[0]
-            elif self.structure_type_id not in structure_types:
-                self.structure_type_id = False
+    @api.depends('company_id')
+    def _compute_structure_type_id(self):
+        StructureType = self.env['hr.payroll.structure.type']
+        for contract in self:
+            contract.structure_type_id = StructureType.search([('country_id', '=', contract.company_id.country_id.id)], limit=1) or\
+                                         StructureType.search([], limit=1)
```
Comment (tivisse) [inline addons/hr_contract/models/hr_contract.py:103]: Searching on a for loop is usually a bad idea. Could be batched IMO.

Comment (dasz-odoo) [reply addons/hr_contract/models/hr_contract.py]: Done.

**Comment by kba-odoo** [comment]
robodoo r+

**Pull request closed without merging**